packages/arb-validator-core/ethbridgetest/message_test.go

```diff
@@ -30,230 +40,350 @@ import (
+		MessageNum: big.NewInt(98742),
+	}
+	bridgeHash, err := tester.EthHash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.Value,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if bridgeHash != msg.CommitmentHash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong hash")
+	}
+
+	messageBridgeHash, err := tester.EthMessageHash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.Value,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if messageBridgeHash != message.DeliveredValue(msg).Hash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong message hash")
+	}
+}
+
+func TestERC20Message(t *testing.T) {
+	msg := message.DeliveredERC20{
+		ERC20: message.ERC20{
+			To:           addr1,
+			From:         addr2,
+			TokenAddress: addr3,
+			Value:        big.NewInt(89735406),
+		},
+		BlockNum:   common.NewTimeBlocks(big.NewInt(87962345)),
+		Timestamp:  big.NewInt(35463245),
+		MessageNum: big.NewInt(98742),
+	}
+	bridgeHash, err := tester.Erc20Hash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.TokenAddress.ToEthAddress(),
+		msg.Value,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if bridgeHash != msg.CommitmentHash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong hash")
+	}
+
+	messageBridgeHash, err := tester.Erc20MessageHash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.TokenAddress.ToEthAddress(),
+		msg.Value,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if messageBridgeHash != message.DeliveredValue(msg).Hash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong message hash")
+	}
+}
+
+func TestERC721Message(t *testing.T) {
+	msg := message.DeliveredERC721{
+		ERC721: message.ERC721{
+			To:           addr1,
+			From:         addr2,
+			TokenAddress: addr3,
+			Id:           big.NewInt(89735406),
+		},
+		BlockNum:   common.NewTimeBlocks(big.NewInt(87962345)),
+		Timestamp:  big.NewInt(35463245),
+		MessageNum: big.NewInt(98742),
+	}
+	bridgeHash, err := tester.Erc721Hash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.TokenAddress.ToEthAddress(),
+		msg.Id,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if bridgeHash != msg.CommitmentHash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong hash")
+	}
+
+	messageBridgeHash, err := tester.Erc721MessageHash(
+		nil,
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.TokenAddress.ToEthAddress(),
+		msg.Id,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+		msg.MessageNum,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if messageBridgeHash != message.DeliveredValue(msg).Hash().ToEthHash() {
+		t.Error("Ethbridge calculated wrong message hash")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [string literal "Ethbridge calculated wrong message hash" appears, at least, 5 times, create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)